src/inc/msquic.h

```diff
@@ -946,6 +946,16 @@ void
 #endif
 #define QUIC_PARAM_GLOBAL_TLS_PROVIDER                  0x0100000A  // QUIC_TLS_PROVIDER
 #define QUIC_PARAM_GLOBAL_STATELESS_RESET_KEY           0x0100000B  // uint8_t[] - Array size is QUIC_STATELESS_RESET_KEY_LENGTH
+#define QUIC_PARAM_GLOBAL_STATELESS_RETRY_CONFIG        0x0100000C  // QUIC_STATELESS_RETRY_CONFIG
+
+//
+// Structure for QUIC_PARAM_GLOBAL_STATELESS_RETRY_CONFIG
+//
+typedef struct QUIC_STATELESS_RETRY_CONFIG {
+    CXPLAT_AEAD_TYPE Algorithm;      // AEAD algorithm for the key
+    uint32_t RotationMs;             // Key rotation interval in milliseconds
+    const uint8_t* Secret;           // Pointer to the base secret
```

Comment:
@nibanks
Is it fine for this to be a pointer (to caller memory), or should the struct have a fixed-sized array for the secret? Or a zero-length array at the end with the secret at the end?

I picked this way so there's fewer copies of the secret in memory (and fewer places the app has to zero out to protect the secret)

Reply:
Instead, I think we should have a variable length buffer at the end of the struct, so it's all contained. That allows the input length to be validated as well.

Reply:
Zero-length arrays are non-standard, and we don't enable the compiler option to use them, so I made an array of size 1, and the caller is expected to allocate enough additional memory beyond the end of the struct to hold the entire secret.

Reply:
Well, we still use them 😄